lhotse/cut/mixed.py

```diff
@@ -688,10 +693,19 @@ def reverb_rir(
         if len(rir_channels) == 1:
             rir_channels = rir_channels * len(self.tracks)
 
-        # NOTE: Currently, if no RIR is provided, this method will generate a
-        # random one for each track in the MixedCut. This is not ideal since the room
-        # configuration for all the RIRs should be the same. But we ignore this for now
-        # since it simplifies the implementation considerably.
+        source_rng_seeds = [source_rng_seed] * len(self.tracks)
```

Comment:
This is the core part of the change. Earlier, a random RIR would be generated when `load_audio()` was called on the reverberated cut for each track. Now, we have 2 RNGs -- one for the room and another for the source location. We use the same room RNG seed for all tracks but different source RNG seed for each track. In this way, we can ensure that all tracks are reverberated with RIRs that are generated using the same room configuration but different source locations.